pandas/errors/__init__.py

```diff
@@ -0,0 +1,63 @@
+# flake8: noqa
+
+""" expose public exceptions & warnings """
+
+from pandas._libs.tslib import OutOfBoundsDatetime
+
+
+class PandasError(Exception):
+    pass
+
+
+class PerformanceWarning(Warning):
+    pass
+
+class AmbiguousIndexError(PandasError, KeyError):
+    pass
+
+
+class UnsupportedFunctionCall(ValueError):
+    pass
+
+
+class UnsortedIndexError(KeyError):
+    """ Error raised when attempting to get a slice of a MultiIndex
+    and the index has not been lexsorted. Subclass of `KeyError`.
+
+    .. versionadded:: 0.20.0
+
+    """
+    pass
+
+
+class ParserError(ValueError):
+    """
+    Exception that is thrown by an error is encountered in `pd.read_csv`
+    """
+    pass
+
+
+class DtypeWarning(Warning):
```

Comment:
I think we should either give some of these CSV specific errors/warnings more specific names, or make the docstrings more generic (i.e., not just referring to `pd.read_csv`).